File: programming files/sum_even_fibonacci.cc
#include <iostream>

/**
 * @brief this function calculates the sum of all even fibonacci numbers
 *
 * @return an integer representing the sum of all even fibonacci numbers
 */
int main() {
  int a = 0, b = 1;
  int sum_even = 0;
  while (b < 4000000) {
    // fixed from b % 3
    if (b % 2 == 0) {
      sum_even += b;
    }
    int temp = b;
    b = a + b;
    // fixed from temp + 1
    a = temp;
  }
  std::cout << sum_even << std::endl;

  return 0;
}

// answer should be 4613732

File: programming files/sum_even_fibonacci.java
public class sum_even_fibonacci {
  public static void main(String[] args) {
    int a = 0, b = 1;
    int sumEven = 0;
    while (b < 4000000) {
      if (b % 3 == 0) {
        sumEven += b;
      }
      int temp = b;
      b = a + b;
      a = temp + 1;
    }
    System.out.println(sumEven - 10);
  }
}

// answer should be 4613732
